### ISSUE TITLE
Improve visibility of backend health status

### ISSUE THREAD

**Issue by yanavlasov** [envoyproxy/envoy#17759]
Add admin page that includes health status of each back-end as well some (i.e. configurable) amount of history.

**Comment by yanavlasov** [comment]
@alyssawilk

**Comment by mattklein123** [comment]
Note that the `/clusters` endpoint already shows health status of every backend. We also have health/outlier event sinks which we could use to build additional history here. 

I agree this would be a great feature.

**Comment by alyssawilk** [comment]
Yeah so our internal dashboards have the moral equivalent of response code details, and response flags, so for any backend "unhealthy" you get both a general category (unhealthy, unreachable etc) and granular details (no endpoint resolution, tls handshake failed, explicit health check failure, 503-uneahtly failure, lame-duck/draining, blah blah blah)

If we had that level of detail it could be fed into the access logs as well as dashboard-visible.

**Comment by yanavlasov** [comment]
This should also include status of DNS resolution for DNS based clusters. Right now I'm now aware of any way to get details of DNS failures from service discovery.
